pygeos/test/test_measurement.py

```diff
@@ -175,3 +175,102 @@ def test_haussdorf_distance_empty():
 def test_haussdorf_distance_densify_empty():
     actual = pygeos.hausdorff_distance(point, empty, densify=0.2)
     assert np.isnan(actual)
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 7, 0), reason="GEOS < 3.7")
+@pytest.mark.parametrize(
+    "geom1, geom2, expected",
+    [
+        # identical geometries should have 0 distance
+        (
+            pygeos.linestrings([[0, 0], [100, 0]]),
+            pygeos.linestrings([[0, 0], [100, 0]]),
+            0,
+        ),
+        # example from GEOS docs
+        (
+            pygeos.linestrings([[0, 0], [50, 200], [100, 0], [150, 200], [200, 0]]),
+            pygeos.linestrings([[0, 200], [200, 150], [0, 100], [200, 50], [0, 0]]),
+            200
+        ),
+        # same geometries but different curve direction results in maximum
+        # distance between vertices on the lines.
+        (
+            pygeos.linestrings([[0, 0], [50, 200], [100, 0], [150, 200], [200, 0]]),
+            pygeos.linestrings([[200, 0], [150, 200], [100, 0], [50, 200], [0, 0]]),
+            200,
+        ),
+        # another example from GEOS docs
+        (
+            pygeos.linestrings([[0, 0], [50, 200], [100, 0], [150, 200], [200, 0]]),
+            pygeos.linestrings([[0, 0], [200, 50], [0, 100], [200, 150], [0, 200]]),
+            282.842712474619
+        ),
+        # example from GEOS tests
+        (
+            pygeos.linestrings([[0, 0], [100, 0]]),
+            pygeos.linestrings([[0, 0], [50, 50], [100, 0]]),
+            70.7106781186548
+        ),
+    ],
+)
+def test_frechet_distance(geom1, geom2, expected):
+    actual = pygeos.frechet_distance(geom1, geom2)
+    assert actual == pytest.approx(expected, abs=1e-12)
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 7, 0), reason="GEOS < 3.7")
+@pytest.mark.parametrize(
+    "geom1, geom2, densify, expected",
+    [
+        # example from GEOS tests
+        (
+            pygeos.linestrings([[0, 0], [100, 0]]),
+            pygeos.linestrings([[0, 0], [50, 50], [100, 0]]),
+            0.001,
+            50
+        ),
+    ],
+)
+def test_frechet_distance_densify(geom1, geom2, densify, expected):
+    actual = pygeos.frechet_distance(geom1, geom2, densify=densify)
+    assert actual == pytest.approx(expected, abs=1e-12)
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 7, 0), reason="GEOS < 3.7")
+@pytest.mark.parametrize(
+    "geom1, geom2",
+    [
+        (point, None),
```

Comment:
I also found that `(point, point)` results in `nan` too.  That seems counterintuitive; shouldn't it be 0?

Reply:
You are right. Actually, these tests should be over line_string to be more precise, as Fréchet distance works over curves and surfaces, not points. Maybe that is also the reason why we are getting a NaN here.

Reply:
I have changed all Fréchet distance tests to use line_strings for now.